sample/testcase.yaml: replace deprecated OVERLAY_CONFIG with EXTRA_CONF_FILE

JarmouniA opened the pull request:
Replace some `OVERLAY_CONFIG` instances that were missed in #79504.

Also, fix incorrect use of `EXTRA_CONF_FILE`, instead of `CONF_FILE`, when the config file actually contains an alternative config & not just additional/modified options.

thedjnK commented on samples/subsys/fs/littlefs/sample.yaml:
```diff
@@ -52,7 +52,7 @@ tests:
     platform_allow: stm32h747i_disco/stm32h747xx/m7
     extra_args:
-      - EXTRA_CONF_FILE=boards/stm32_blk.conf
+      - EXTRA_CONF_FILE=overlay-stm32_blk.conf
       - CONF_FILE=prj_blk.conf
```
No, do NOT do this

JarmouniA replied:
@thedjnK the majority of overlay config files already have the prefix `overlay-`, this change just makes things consistent & makes it less likely for unfamiliar users to confuse base config files with overlay ones.

thedjnK replied:
Overlay isn't even right, overlay is a devicetree term, it has no relation to Kconfig at all

JarmouniA replied:
We can debate the terminology, but it's still an `overlay` file that is applied on top of the base one & overrides it.
We can replace with `extra` if that's better.

thedjnK replied:
It's not an overlay, we will do no replacing of sorts

JarmouniA replied:
> No, do NOT do this

@thedjnK I moved that file because it applies to multiple boards & thus does not belong in `boards`.

JarmouniA replied:
> Then why did you add it wrong in the first place?

I was new to Zephyr :)

> it makes no sense to break everyone's workflow in existing samples which use those names, but new samples/tests should not

I agree that the renaming is likely to cause breakage out-of-tree... will remove, thanks for your review!

thedjnK reviewed (changes requested):
I really do not see the value in any of these renames

JarmouniA committed:
samples: subsys: shell: devmem_load: fix incorrect use of OVERLAY_CONFIG

Replace OVERLAY_CONFIG with CONF_FILE since the file in question
contains an alternative configuration to the one in prj.conf
as clearly stated in the sample doc
'If you use poll you should also use `prj_poll.conf` instead of
`prj.conf`.'

Signed-off-by: Abderrahmane Jarmouni <git@jarmouni.me>

JarmouniA committed:
tests: testcase.yaml: replace deprecated OVERLAY_CONFIG w\ EXTRA_CONF_FILE

Replace some OVERLAY_CONFIG that were missed in a pervious PR.

Signed-off-by: Abderrahmane Jarmouni <git@jarmouni.me>

JarmouniA committed:
tests: subsys: pm: power_mgmt: fix incorrect use of OVERLAY_CONFIG

Replace OVERLAY_CONFIG with CONF_FILE since no-device-pm.conf
contains an alternative configuration to the one in prj.conf

Signed-off-by: Abderrahmane Jarmouni <git@jarmouni.me>

JarmouniA committed:
samples: subsys: fs: littlefs: move stm32_blk.conf

Move stm32_blk.conf ouside of 'boards' as this folder is meant to
contain only board-specific devicetree overlays and Kconfig fragments.

Signed-off-by: Abderrahmane Jarmouni <git@jarmouni.me>

JarmouniA committed:
samples: sample.yaml: replace deprecated OVERLAY_CONFIG w\ EXTRA_CONF_FILE

Replace some OVERLAY_CONFIG that were missed in a pervious PR.

Signed-off-by: Abderrahmane Jarmouni <git@jarmouni.me>
